lib/util/http/jira.ts

```diff
@@ -0,0 +1,22 @@
+import type { HttpOptions, HttpResponse, InternalHttpOptions } from './types';
```

Comment:
can we extract this to be merged first, even it's not yet used then.

Reply:
Certainly - there is lots in this WIP branch that will need to be teased out and added in chunks

https://github.com/renovatebot/renovate/pull/21056 for the jira http util